### DISCUSSION TITLE
Route Model Binding (or not)

### DISCUSSION THREAD

**Discussion by wrurik** [spatie/laravel-query-builder#613] [Q&A]
Hello,

We use this package for building an API. We use Laravel Resources for the response of the API and all relations are conditionally appended to the Resources like this:
```
'relation' => RelationResource::make($this->whenLoaded('relation')),
```

I was trying to setup the Spatie query builder to be used when resolving RouteModelBinding so I can request certain relations on a model when fetching a single model or even when updating a model.

I kinda got it working by placing the following in my `RouteServiceProvider`:

```
Route::bind('order', function ($value) {
  return (new OrderQuery)->where('id', '=', $value)->firstOrFail();
});
```

This works when `order` is the only thing being resolved from the url:
`/api/orders/{order}`

But when I have a nested resource this fails because the filters and relations in the request (intended for `order-lines`) are applied when resolving the `order` as well:
`/api/orders/{order}/order-lines`

This made me wonder how other people use this package. Do you use the Spatie Query Builder at all for routes other than the index? And if not, how do you determine the relations that should be loaded in these other methods?

**Comment by AlexVanderbist** [comment]
Hi! I've never used the package as a route model binding but it's definitely an interesting use case. I think most people indeed just create the `QueryBuilder` instance in the `index` method of their controllers.

However, I have used an extended `QueryBuilder` class (like your `OrderQuery`) to re-use on both the `index` and `show` routes of an API controller. That way the front-end has access to the same includes on a single resource.

Hopefully this answers your question, good luck!
